src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Datagrid.js

```diff
@@ -63,23 +74,32 @@ export default class Datagrid extends React.PureComponent<Props> {
         selected ? store.selectEntirePage() : store.deselectEntirePage();
     };
 
+    handleAdapterChange = (adapter: string) => {
+        this.setCurrentAdapter(adapter);
+        this.props.store.updateLoadingStrategy(this.currentAdapter.adapter.getLoadingStrategy());
+    };
+
     render() {
         const {
             store,
             onItemClick,
+            adapters,
         } = this.props;
         const page = store.getPage();
         const pageCount = store.pageCount;
-        const {
-            Adapter,
-            paginationType,
-        } = this.getAdapterConfig();
+        const Adapter = this.currentAdapter.adapter;
+
         return (
             <div>
+                <AdapterSwitch
+                    adapters={adapters}
+                    currentAdapter={this.currentAdapter.key}
+                    onAdapterChange={this.handleAdapterChange}
+                />
                 {this.props.store.loading && !this.props.store.appendRequestData
                     ? <Loader />
                     : <PaginationDecorator
```

Comment:
Maybe we should also add a TODO here, saying that we want to get rid of this decorator. Not that we forget it in the end 😃